planner/core/rule_build_key_info.go

```diff
@@ -273,7 +293,7 @@ func (is *LogicalIndexScan) BuildKeyInfo(selfSchema *expression.Schema, childSch
 		if path.IsTablePath() {
 			continue
 		}
-		if newKey := checkIndexCanBeKey(path.Index, is.Columns, selfSchema); newKey != nil {
+		if _, newKey := checkIndexCanBeKey(path.Index, is.Columns, selfSchema); newKey != nil {
```

Comment:
We should maintain this `UniqueKeys` for other operators as well, in their own `BuildKeyInfo` implementations, otherwise this info would be lost if the plan tree is like `Selection -> xxx... -> DataSource`.

Reply:
done